### ISSUE TITLE
P2P Connectivity issues

### ISSUE THREAD

**Issue by arkpar** [paritytech/parity#4541]
Originally reported by @lgpawel
Log available at: https://mega.nz/#!4lYT1DbA!GcfbaFEemd6NX9fiFR1jklaOSfMAZEWOyA2FhlroD74 
Parity refuses to connect to peers and the session counter does not account for some deleted sessions.

There seems to be 2 separate issues here.
1. Node is sending out packets over the protocol limit of 16MB. Huge blocks perhaps? This should be handled better and such packets must not be allowed out.
2. Because of 1. there's a loop for disconnecting and re-connecting nodes over and over again. This causes the connection counter to get out of sync somehow. This is probably the same issue as #4261

**Comment by arkpar** [comment]
@lgpawel could you try building the `network-testing` branch and see if it fixes the issue using exactly the same setup?

**Comment by lgpawel** [comment]
I'll try at some point, but can't really promise when that will be. When I do, is `-l network=trace,rpc=trace` a right log level to use or should I include something further?

And I'd be really surprised if the issue stemmed from huge blocks, as everything seems to indicate that the blocks are empty instead, eg. `grep Imported from3to198sessions.log` reports all blocks at 0 tx, 0.52 kB.

**Comment by arkpar** [comment]
Hm, would be good to know where oversized packets are coming from. Transactions maybe?
`2017-02-12 19:58:48 UTC IO Worker #2 TRACE network  35: Sending 17116544 bytes`
`-l network=trace,rpc=trace,sync=trace` would provide a better clue.

**Comment by lgpawel** [comment]
Transactions are much more plausible. I had scripts calling them at a rate of 5 per second for this node. But the transactions themselves are quite small, at least when mined in blocks - other logs for the same setup indicate that each of them occupies some 140B when mined:
```
2017-02-10 17:54:55 UTC Verifier #3 INFO import  Imported #991 c430…ce80 (80 txs, 11.40 Mgas, 5054.34 ms, 12.51 KiB)
```
Assuming that they use comparable space when exchanged between queues of the nodes, it'd mean a blob of 16MB would have over 100k of them (which is not impossible though). It seems that they linger in the queue and for some reason never get mined. This in itself is certainly a problem with my setup, but I guess it shouldn't cause disconnects etc.

**Comment by tomusdrw** [comment]
We broadcast transactions in batches of 64 per packet (in recent versions), so they would have to be much larger.

**Comment by lgpawel** [comment]
The version is `Parity/v1.6.0-unstable-0bf9f6c-20170203/x86_64-linux-gnu/rustc1.15.0`, for reference. How do I reliably measure the size of the transaction while stored in the queue/exchanged between nodes?

**Comment by 5chdn** [comment]
This issue is labelled with _bug_: The client fails to follow expected behavior; and is inactive for several months. It's assigned to @arkpar and linked to the _patch_ milestone.

@lgpawel do you still have connectivity issues with 1.7.0-unstable (latest master)?

Could anyone get back to the last question by @lgpawel (previous comment) within 7 days and state if this is still relevant? Please close this if fixed otherwise.

**Comment by arkpar** [comment]
The original issue is fixed in #4563
